src/appshell/view/dockwindow/dockwindow.h

```diff
@@ -91,9 +101,13 @@ private slots:
     void showPage(DockPage* page);
     void adjustPanelsSize(DockPage* page);
 
+    QMenu* makeMenu(const uicomponents::MenuItem& menuItem) const;
+    QAction* makeAction(const uicomponents::MenuItem& menuItem) const;
+
     QMainWindow* m_window = nullptr;
     EventsWatcher* m_eventsWatcher = nullptr;
     QString m_title;
+    uicomponents::MenuItemList m_menues;
```

Comment:
I think it's better for dock to use its types rather than uicomponents so that it remains independent (dock will be easy to replace)